lib/ocpp/v2/charge_point.cpp

```diff
@@ -589,16 +589,19 @@ void ChargePoint::initialize(const std::map<int32_t, int32_t>& evse_connector_st
         this->callbacks.stop_transaction_callback, this->registration_status, this->upload_log_status,
         this->upload_log_status_id);
 
-    if (device_model->get_optional_value<bool>(ControllerComponentVariables::V2XChargingCtrlrAvailable)
-            .value_or(false)) {
+    const auto v2x_available_map =
+        device_model->get_all_optional_values_for_each_evse<bool>("V2XChargingCtrlr", V2xComponentVariables::Available);
```

Comment:
To me it lools like you dont need `get_all_optional_values_for_each_evse` and could use something like this instead:

```cpp
bool any_v2x_available = std::any_of(
    evse_connector_structure.begin(),
    evse_connector_structure.end(),
    [this](const auto& entry) {
        const auto& [evse, connectors] = entry;
        return this->device_model
            ->get_optional_value<bool>(
                V2xComponentVariables::get_component_variable(evse, V2xComponentVariables::Available)
            )
            .value_or(false);
    }
);
``